pkgs/development/compilers/llvm/common/lldb.nix

```diff
@@ -160,6 +160,9 @@ stdenv.mkDerivation (rec {
     ln -s $out/bin/${if (lib.versionOlder release_version "12") then "llvm-vscode" else "lldb-vscode"} $out/share/vscode/extensions/llvm-org.lldb-vscode-0.1.0/bin
   '';
 
+  # TODO: Hack to make dynamic linking work again. Find a better way to do this using the build-system machinery
+  postFixup = if (stdenv.isLinux && lib.versionAtLeast release_version "16") then "patchelf --add-rpath ${libclang.lib}/lib $out/bin/.lldb-wrapped" else null;
```

Comment:
this needs to also patch `lldb-server`, `lldb-instr`, `lldb-argdumper`, `lldb-vscode` and potentially others